gradle.properties

```diff
@@ -1,3 +1,21 @@
+# Project-wide Gradle settings.
+
+# IDE (e.g. Android Studio) users:
+# Gradle settings configured through the IDE *will override*
+# any settings specified in this file.
+
+# For more details on how to configure your build environment visit
+# http://www.gradle.org/docs/current/userguide/build_environment.html
+
+# Specifies the JVM arguments used for the daemon process.
+# The setting is particularly useful for tweaking memory settings.
+org.gradle.jvmargs=-Xmx1536m
```

Comment:
@EricKuck Good catch, I was wondering about something similar when I first saw this. However, it turned out with more recent versions of the Android build toolchain, dex will run out of process (slowing down the build) if the maximum heap space is not set to at least 1536 MB. The jvmargs line I added is now **the default setting** when creating a new Android project.
Running Gradle *without* this parameter will output the following warning during the build:
```
Running dex as a separate process.

To run dex in process, the Gradle daemon needs a larger heap.
It currently has 1024 MB.
For faster builds, increase the maximum heap size for the Gradle daemon to at least 1536 MB.
To do this set org.gradle.jvmargs=-Xmx1536M in the project gradle.properties.
For more information see https://docs.gradle.org/current/userguide/build_environment.html
```
Note that `Xmx` only specifies the *maximum* memory allocation, not the initial amount (which would be `Xms`).